### DISCUSSION TITLE
What environment variables are required for brew to install successfully?

### DISCUSSION THREAD

**Discussion by jwodder** [Homebrew/discussions#1313] [Everyday usage] [answered]
I'm writing a Python script that calls `brew`, and I'm testing it using [tox](https://tox.rtfd.io).  Unfortunately, when run inside tox, `brew install` fails with "Warning: The post-install step did not complete successfully" when trying to install anything, yet the same code works when run outside tox.  I believe this is due to tox [filtering environment variables](https://tox.readthedocs.io/en/latest/config.html#conf-passenv); apparently, something that brew needs is not getting passed through.  What are the minimal environment variables that must be set in order for brew to be able to install formulae successfully?

**Comment by SMillerDev** [comment]
Nothing that I know of. But post-install steps could require specific variables. There is no way to know what is going wrong without logs though.

**Reply by jwodder** [reply]
Which logs, exactly?  I tried running `brew install` with `--verbose`, and all I can tell is that the failure is occuring on or after symlinking.  [Here's a GitHub Actions run of an MVCE.](https://github.com/jwodder/test/runs/2381665812?check_suite_focus=true)

**Reply by carlocab** [reply]
What happens when you run `brew install git-annex` directly in CI, and not through python/tox?

**Reply by SMillerDev** [reply]
Try running it with `--debug`

**Reply by jwodder** [reply]
@carlocab `brew install git-annex` works fine in GitHub Actions when run either directly or via a Python script not run under tox.

**Reply by jwodder** [reply]
@SMillerDev With `--debug`, I see "`Errno::EACCES: Permission denied @ dir_s_mkdir - /Library/Logs/Homebrew`", which suggests that the unset `HOME` envvar is being replaced with an empty string.  Indeed, if I add `passenv = HOME` to the tox config, the installation succeeds.

**Comment by jwodder** [comment, marked as answer]
It turns out that the failure was due to the `HOME` envvar not being set; configuring tox to pass the envvar through causes the tests to succeed.  I leave it up to the maintainers whether this should be considered a bug in Homebrew.
